homeassistant/components/esphome/sensor.py

```diff
@@ -52,6 +52,8 @@ def _state(self) -> Optional[SensorState]:
     @property
     def icon(self) -> str:
         """Return the icon."""
+        if self._static_info.icon == "":
```

Comment:
If a device class is set, integration should not provide an icon.

```suggestion
        if not self._static_info.icon or self._static_info.device_class:
```